src/main/scala/io/circe/yaml/Printer.scala

```diff
@@ -27,17 +28,49 @@ final case class Printer(
   version: YamlVersion = YamlVersion.Auto
 ) {
 
+  def containsBuggyChars(json: Json): Boolean = {
+    def isBad(s: String): Boolean =
+      s.indexOf('\u0085') >= 0 || s.indexOf('\ufeff') >= 0
+
+    @tailrec def loop(js: List[Json]): Boolean =
+      js match {
+        case h :: t =>
+          h match {
+            case Json.JArray(js) => loop(js.toList ::: t)
+            case Json.JObject(o) => loop(o.values.toList ::: t)
+            case Json.JString(s) if isBad(s) => true
+            case _ => loop(t)
+          }
+        case Nil =>
+          false
+      }
+    loop(json :: Nil)
+  }
+
   def pretty(json: Json): String = {
     val rootTag = yamlTag(json)
     val writer = new StringWriter()
-    val serializer = new Serializer(new Emitter(writer, options), new Resolver, options, rootTag)
+    val opts = if (containsBuggyChars(json)) safeOptions else options
+    val serializer = new Serializer(new Emitter(writer, opts), new Resolver, opts, rootTag)
     serializer.open()
     serializer.serialize(jsonToYaml(json))
     serializer.close()
     writer.toString
   }
 
-  private lazy val options = {
+  private lazy val options = makeOptions()
+
+  // these options are needed so that 0x0085 and 0xFEFF render
+  // correctly. these settings are just to work around a bug in
+  // SnakeYAML -- we only use them if we see those characters.
+  private lazy val safeOptions = {
+    val opts = makeOptions()
+    opts.setDefaultScalarStyle(DumperOptions.ScalarStyle.DOUBLE_QUOTED)
```

Comment:
Couldn't we call these on the user's `options` rather than creating a new one from scratch?